### ISSUE TITLE
grpc-web-root-path setting ignored in ARGOCD_OPTS

### ISSUE THREAD

**Issue by awasilyev** [argoproj/argo-cd#6822]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

grpc-web-root-path setting ignored in ARGOCD_OPTS

**To Reproduce**

```
$ export ARGOCD_OPTS="--grpc-web-root-path=/argocd"
$ argocd app list
FATA[0001] rpc error: code = Internal desc = transport: received the unexpected content-type "text/html"
$ argocd --grpc-web-root-path=/argocd app list
NAME                                    CLUSTER                NAMESPACE          PROJECT  STATUS  HEALTH   SYNCPOLICY  CONDITIONS  REPO                                                PATH                                     TARGET
```

**Expected behavior**

list of apps

**Screenshots**

**Version**

```shell
argocd: v2.0.5+4c94d88
  BuildDate: 2021-07-22T20:54:35Z
  GitCommit: 4c94d886f56bcb2f9d5b3251fdc049c2d1354b88
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: darwin/amd64```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by alexmt** [comment]
Thank you @charles-woshicai , looking forward for the PR!

**Comment by awasilyev** [comment]
got it working with
export ARGOCD_OPTS="--grpc-web-root-path /argocd"

**Comment by charles-woshicai** [comment]
> got it working with
> export ARGOCD_OPTS="--grpc-web-root-path /argocd"

Cool, seems like pkg `cobra` doesn't receive input arguments in format `foo=bar` but `foo bar` if so.
I will test later.

**Comment by charles-woshicai** [comment]
It's true that the argocd client doesn't recognize opts in format `foo=bar`, and this is caused by the pkg `shellquota` that argocd client used to load flags. 
https://github.com/argoproj/argo-cd/blob/bcfc112d828476270c006a349be4c14228336be5/util/config/env.go#L24

pkg version: `github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51`

I write some simple code to test this:
```go
	Run: func(cmd *cobra.Command, args []string) {
		input, err := shellquote.Split(opts)
		if err != nil {
			panic(err)
		}
		log.Info().Interface("opts", input).Msg("test")
	},
```

Here is the test results:
```shell
➜  test git:(master) ✗ go run main.go test1 --opts --grpc-web-root-path=/argocd  
{"level":"info","opts":["--grpc-web-root-path=/argocd"],"time":"2021-08-02T23:12:25+08:00","message":"test"}
➜  test git:(master) ✗ go run main.go test1 --opts "--grpc-web-root-path /argocd"
{"level":"info","opts":["--grpc-web-root-path","/argocd"],"time":"2021-08-02T23:12:29+08:00","message":"test"}
```

As you can see,  pkg`shellquota` doesn't recognize `=` so the opt in format `foo=bar` is ignored.

**Comment by charles-woshicai** [comment]
@alexmt Please make a decision whether argo-cd client support opts in format `foo=bar`.

**Comment by alexmt** [comment]
@charles-woshicai  I agree - we should support `foo=bar` format as well. Thank you for PR!
